buildpacks/nodejs-function-invoker/src/main.rs

```diff
@@ -0,0 +1,140 @@
+use libcnb::data::build_plan::BuildPlanBuilder;
+use libcnb::data::launch::{Launch, ProcessBuilder};
+use libcnb::data::{layer_name, process_type};
+use libcnb::detect::{DetectContext, DetectResult, DetectResultBuilder};
+use libcnb::generic::GenericPlatform;
+use libcnb::{buildpack_main, Buildpack};
+#[cfg(test)]
+use libcnb_test as _;
+use libherokubuildpack::{log_error, log_header, log_info};
+use serde::Deserialize;
+use thiserror::Error;
+#[cfg(test)]
+use ureq as _;
+
+mod function;
+mod layers;
+
+pub struct NodeJsInvokerBuildpack;
+
+#[derive(Deserialize, Debug)]
+pub struct NodeJsInvokerBuildpackMetadata {
+    pub runtime: NodeJsInvokerBuildpackRuntimeMetadata,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct NodeJsInvokerBuildpackRuntimeMetadata {
+    pub package: String,
+}
+
+impl Buildpack for NodeJsInvokerBuildpack {
+    type Platform = GenericPlatform;
+    type Metadata = NodeJsInvokerBuildpackMetadata;
+    type Error = NodeJsInvokerBuildpackError;
+
+    fn detect(&self, context: DetectContext<Self>) -> libcnb::Result<DetectResult, Self::Error> {
+        is_function(&context.app_dir)
+            .then(|| {
+                DetectResultBuilder::pass()
+                    .build_plan(
+                        BuildPlanBuilder::new()
+                            .requires("node")
+                            .requires("nodejs-function-invoker")
+                            .provides("nodejs-function-invoker")
+                            .build(),
+                    )
+                    .build()
+            })
+            .unwrap_or_else(|| DetectResultBuilder::fail().build())
+    }
+
+    fn build(&self, context: BuildContext<Self>) -> libcnb::Result<BuildResult, Self::Error> {
+        log_header("Heroku Node.js Function Invoker Buildpack");
+
+        log_info("Checking for function file");
+        get_main(&context.app_dir).map_err(NodeJsInvokerBuildpackError::MainFunctionError)?;
+
+        context.handle_layer(
+            layer_name!("runtime"),
+            RuntimeLayer {
+                package: context
+                    .buildpack_descriptor
+                    .metadata
+                    .runtime
+                    .package
+                    .clone(),
+            },
+        )?;
+
+        BuildResultBuilder::new()
+            .launch(
+                Launch::new().process(
+                    ProcessBuilder::new(process_type!("web"), "sf-fx-runtime-nodejs")
+                        .args(vec![
+                            "serve",
+                            &context.app_dir.to_string_lossy(),
+                            "--workers",
+                            "2",
+                            "--host",
+                            "::",
+                            "--port",
+                            "${PORT:-8080}",
+                            "--debug-port",
+                            "${DEBUG_PORT:-}",
+                        ])
+                        .default(true)
```

Comment:
Nit: Even though it is the default, we could add `.direct(false)` here to explicitly show we expect this to be run via `bash`.

Reply:
Fixed in https://github.com/heroku/buildpacks-nodejs/pull/224/commits/5e1b6e50bcdb49e58c7f963ef8f43883b142f1e0